### PR TITLE
DEV ONLY - NEO-2402 // `LeftNav` -> `SideNavigation`

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to Left Nav storybook](https://deploy-preview-549--neo-react-library-storybook.netlify.app/?path=/docs/components-left-navigation--docs) | [Link to Side Navigation storybook](https://deploy-preview-549--neo-react-library-storybook.netlify.app/?path=/docs/components-side-navigation--docs)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [ ] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

Renamed, added a backwards compatibility test, added "DEPRECATED" notes in component JSDocs and stories, updated storybook pages to use latest syntax.

See CodeRabbit notes for more details. 

## Summary by CodeRabbit

- **New Features**
	- Introduced the `SideNavigation` component to replace the previous `LeftNav` component, enhancing the navigation experience.
	- Added a comprehensive suite of Storybook stories for the `SideNavigation` component, showcasing various configurations.

- **Bug Fixes**
	- Improved accessibility compliance with a focus on ARIA standards in the `SideNavigation` component.

- **Refactor**
	- Deprecated the `LeftNav` component, consolidating its functionality into the new `SideNavigation` component.
	- Updated all related components and tests to align with the new `SideNavigation` structure.

- **Chores**
	- Removed outdated `LeftNav` stories and tests, streamlining the codebase.

### CODE REVIEW

File: src/components/TopNav/TopNav.stories.tsx [hunk 1]
```diff
@@ -55,47 +55,59 @@ NavigationToggle.args = {
...
 			<>
 				<Story args={{ ...args, menuToggleBtn: navMenuToggleWithHandler }} />
-				{displayLeftNav && (
+				{displaySideNav && (
 					<div
 						className={
-							displayLeftNav
+							displaySideNav
 								? "neo-slide neo-slide--in-left neo-leftnav--collapsible"
 								: "neo-leftnav--collapsible neo-slide neo-slide--out-left"
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/TopNav/TopNav.stories.tsx:74]: @avaya-dux/dux-devs CodeRabbit has made a _bunch_ of helpful notes. Most of them are around updating CSS class names, so I'm not implementing them, but all the same, a list like this is shockingly helpful and thought that you two would be interested to see it.

File: src/components/SideNavigation/SideNavigation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import { SideNavigation } from ".";
+import {
+	CategoryGroups,
+	CategoryGroupsWithIcons,
+	Default,
+	DoesNotConflictWithOtherNavs,
+} from "./SideNavigation.stories";
+
+describe("Side Navigation", () => {
+	window.HTMLElement.prototype.scrollIntoView = vi.fn();
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/SideNavigation/SideNavigation.test.tsx:18]: thanks! I've made a note and will ignore this for the current PR

File: src/components/SideNavigation/SideNavigation.test.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+				expect(container).not.toBe(null);
+			});
+
+			it("passes basic axe compliance", async () => {
+				const { container } = renderResult;
+				const results = await axe(container);
+				expect(results).toHaveNoViolations();
+			});
+		});
+	});
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/SideNavigation/SideNavigation.test.tsx]: implemented, thanks!

File: src/components/LeftNav/LeftNav.tsx [hunk 4]
```diff
@@ -1,92 +1,15 @@
...
-				<div id={navId} className="neo-leftnav--wrapper">
-					<nav className="neo-leftnav" aria-label={ariaLabel} {...rest}>
-						<ul className="neo-leftnav__nav">{children}</ul>
-					</nav>
-				</div>
-			</LeftNavContext.Provider>
-		</RovingTabIndexProvider>
-	);
+export const LeftNav = (props: SideNavigationProps) => {
+	return <SideNavigation {...props} />;
```
Comment (yangAtSpoken) [inline src/components/LeftNav/LeftNav.tsx:12]: I propose freezing the behavior of LeftNav as it is at this point in time.  This line of code says that LeftNav will evolve together with SideNavigation.

File: src/components/SideNavigation/SideNavigation.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+				</SideNavigation.LinkItem>
+				<SideNavigation.LinkItem href="http://bidi.com">
+					Bi-directional
+				</SideNavigation.LinkItem>
+			</SideNavigation.NavCategory>
+		</SideNavigation>
+	),
+};
+
+export const DoesNotConflictWithOtherNavs: Story = {
```
Comment (joe-s-avaya) [inline src/components/SideNavigation/SideNavigation.stories.tsx:164]: @enrique-prado please review this story

File: src/components/Layout/AppLayout.stories.tsx [hunk 6]
```diff
@@ -2,7 +2,7 @@ import type { Meta } from "@storybook/react";
 import type { ReactNode } from "react";
 
 import { Icon } from "components/Icon";
-import { LeftNav } from "components/LeftNav";
+import { SideNavigation } from "components/SideNavigation";
```
Comment (yangAtSpoken) [inline src/components/Layout/AppLayout.stories.tsx:5]: Good to switch.

File: src/components/LeftNav/LeftNav.tsx (hunk 4 above)
Comment (joe-s-avaya) [reply src/components/LeftNav/LeftNav.tsx:12]: @yangAtSpoken good point, now that I've thought about it a bit, I agree. I'll revert this change and then ping you once it's done.

**Review by yangAtSpoken — APPROVED** [review]
I am a bit concerned that the if the user wants to stay with the current behavior of LeftNav, they will not be able to do so since LeftNav will grow  together with SideNavigation.  But who knows, maybe they all come onboard the new ship quickly.

**Comment by joe-s-avaya** [comment]
@avaya-dux/dux-devs I have taken Yang's advice and I no longer link `LeftNav` to `SideNavigation`. I have kept the updates to the `LeftNav` tests and stories, and the update to the JSDocs to mark it as deprecated so that it is obvious that devs should no longer use it.

**Pull request merged**